registry/proxy/proxyregistry.go

```diff
@@ -211,6 +211,15 @@ func (pr *proxyingRegistry) BlobStatter() distribution.BlobStatter {
 	return pr.embedded.BlobStatter()
 }
 
+type Closer interface {
+	// Close release all resources used by this object
+	Close() error
+}
```

Comment:
Slightly wondering if we must define our own interface for this, or if `io.Closer` (same signature) would do 🤔 

However, looking at the changes in this PR, it looks like we're differentiating the "proxy-registry" here. Usually I'm for "small interfaces", but I did notice there's a TODO for the `Repository` interface as a whole, so wondering if we should go that direction instead, and expand the `Repository` interface to require a `Close()` method? https://github.com/distribution/distribution/blob/bc6e81e1b9a8017a66fc56a55913b92930941feb/registry.go#L116-L118

Reply:
> Slightly wondering if we must define our own interface for this, or if io.Closer (same signature) would do 🤔

I was thinking about this exactly, but I figured at the time it'd be semantically confusing 🤷‍♂️

Reply:
> Slightly wondering if we must define our own interface for this, or if io.Closer (same signature) would do 🤔

I agree with @milosgajdos. I also prefer to do not use `io.Closer` as the registry is not an IO resource, it is a different abstraction.

---

@thaJeztah , thanks for noticing the TODO. Actually, we don't have a strong opinion on the topic. We are at most familiar with the proxy registry codebase and we are using the proxy scenario only. With this PR our main goal is to fix https://github.com/distribution/distribution/issues/2374 and not do design/redesign the `Repository` interface. If you as a maintainer have a strong opinion that this should instead added in the Repository interface instead, we can do it. I think we are not that familiar with all of the existing codebase to decide whether the other `Repository` implementations need to close as well. The proxyRegistry needs such Close method for sure to be able to stop the scheduler (which saves the `scheduler-state.json` file).